alacritty_terminal/src/term/search.rs

```diff
@@ -201,8 +222,29 @@ impl<T> Term<T> {
         end: Point,
         direction: Direction,
         anchored: bool,
-        regex: &impl Automaton,
+        regex: &mut LazyDfa,
     ) -> Option<Point> {
+        match self.regex_search_internal(start, end, direction, anchored, regex) {
+            Ok(regex_match) => regex_match,
+            Err(err) => {
+                log::warn!("Regex exceeded complexity limit");
```

Comment:
Maybe we should say which regex somehow? Also, import `warn!` and `debug!`.

Reply:
I considered that but it's only really viable if we store the pattern in the dfa and I didn't want to do that just for log messages. It should be pretty clear anyway since it's triggered by user actions usually (and should be very rare).